src/Ledger/Fees.lagda

```diff
@@ -71,16 +83,20 @@ scriptsCost pp scSz with (PParams.refScriptCostStride pp)
 The function \scriptsCost{} (Fig.~\ref{fig:scriptsCost}) calculates
 the fee for reference scripts in the transaction using a function that
 is piece-wise linear in the size, where the linear constant multiple
-grows with each \refScriptCostStride{} bytes. Thus, the \scriptsCost{}
-function depends on the \AgdaFunction{scriptsTotalSize} function,
-which returns an integer that is the total size of the reference
-script in bytes, as well as the following protocol parameters:
+grows with each \AgdaFunction{refScriptCostStride} bytes.
+%
+This function takes as input the total size of the reference script in
+bytes. The function \AgdaFunction{scriptsTotalSize} is used to
```

Comment:
Where is `scriptsTotalSize` defined?

Reply:
It doesn't seem to exist. I think it may refer to `refScriptsSize` going by the explanation, so maybe there was some renaming going on?

This explanation is also quite confusing, I'm not sure what a function depending on another function is supposed to mean. I'd just write something like '\scriptsCost{} expects the sum of sizes of reference scripts as computed by \refScriptsSize{} as its input'.

Reply:
Now that I look at it, it's also confusing to call these arguments protocol parameters (below). They're currently hardcoded and may become protocol parameters in the future, if we don't manage to get rid of that function.

Reply:
I see. Maybe it would be best to redefine `scriptsCost` such that all those are given as arguments. WDYT?

Reply:
That just shifts the problem to the call site, but they're still going to be hard coded somewhere. Having them at least be defined right next to where they are used makes it more local.

I guess we could have some type that holds these things, but I think this is the only instance we currently have of these magic numbers. Another option would be to add them to the `GlobalConstants`, but these also have a very specific meaning (these are set by the 'genesis file' in the implementation). So all in all I don't think this can really be improved.